echopop/utils/batch_load.py

```diff
@@ -0,0 +1,608 @@
+    """
+
+    # Get the export dataframe columns present in the `ECHOVIEW_EXPORT_MAP` API
+    valid_columns = {
+        old: new_info
+        for old, new_info in ECHOVIEW_EXPORT_MAP.items()
+        if old in export_dataframe.columns
+    }
+
+    # Drop unnecessary columns
+    export_dataframe_filtered = export_dataframe.copy().filter(valid_columns.keys())
+
+    # Validate the datatypes
+    for old, new_info in valid_columns.items():
+        # ---- Get expected datatype for the matched key name
+        expected_dtype = new_info["type"]
+        # ---- Transform the datatype, if needed
+        export_dataframe_filtered[old] = export_dataframe_filtered[old].astype(expected_dtype)
+
+    # Transform the column names
+    # ---- Get the new names
+    rename_dict = {old: new_info["name"] for old, new_info in valid_columns.items()}
+    # ---- Rename the columns
+    return export_dataframe_filtered.rename(columns=rename_dict)
+
+
+def get_transect_numbers(export_files: list, transect_pattern: str, file_directory: str):
+    """
+    Extract transect numbers from export file strings.
+    """
+
+    # Validate `transect_pattern`
+    if not isinstance(transect_pattern, str):
+        raise TypeError(
+            f"Input value for `transect_pattern` ({transect_pattern}) must be a regular expression "
+            f"or `str`."
+        )
+
+    # Helper function for creating a filename-transect association reference dictionary
+    def extract_transect_number(file, file_pattern=transect_pattern):
+        # ---- Get base filename
+        filename = os.path.basename(file)
+        # ---- Parse string
+        parsed_transect = re.search(file_pattern, filename)
+        # ---- Extract transect number, if present
+        if parsed_transect:
+            try:
+                return int(parsed_transect.group(1)), file
+            # ---- Return None if the transect number is not parseable
+            except ValueError:
+                return None, file
+        else:
+            return None, file
+
+    # ---- Generate filename-transect association reference dictionary
+    transect_reference = {file: num for num, file in map(extract_transect_number, export_files)}
+
+    # Evaluate whether any files failed to be parsed
+    unparsed_transects = [file for file, num in transect_reference.items() if num is None]
+    # ---- If some are missing
+    if unparsed_transects:
+        # ---- Join files on multiple lines
+        unparsed_transects_formatted = "\n".join(unparsed_transects)
+        if len(unparsed_transects) > 10:
+            # ---- Get number of unparsed strings
+            unparsed_n = len(unparsed_transects)
+            # ---- Raise error
+            raise ValueError(
+                f"Transect numbers could not be parsed from {unparsed_n} files in {file_directory}."
+                f" The first 10 files include:\n {unparsed_transects_formatted}"
+            )
```

Comment:
This will still print out all of the files, right? since you did not do any selection in `unparsed_transects_formatted = "\n".join(unparsed_transects)`.

Why not just get those into a list and print the list, and let the IDE do the job of cutting off/hiding the outputs by their rules?